components/resource/scenemanager.cpp

```diff
@@ -515,9 +515,13 @@ namespace Resource
             // share state
             // do this before optimizing so the optimizer will be able to combine nodes more aggressively
             // note, because StateSets will be shared at this point, StateSets can not be modified inside the optimizer
-            mSharedStateMutex.lock();
-            mSharedStateManager->share(loaded.get());
-            mSharedStateMutex.unlock();
+            static bool allowOptimizations = true;
+            if (allowOptimizations)
```

Comment:
If you do not want to merge nodes, you can just disable this feature via OPENMW_OPTIMIZE environmental variable.